source/PlayerInfo.cpp

```diff
@@ -2455,6 +2461,40 @@ void PlayerInfo::StepMissions(UI *ui)
 				if(ship->IsDestroyed())
 					mission.Do(ShipEvent(nullptr, ship, ShipEvent::DESTROY), *this, ui);
 	
+	// Move all invalid missions to the end of the mission list.
+	missions.sort([] (const Mission &lhs, const Mission &rhs)
+	{
+		return lhs.IsValid();
+	});
+	auto mit = find_if(missions.begin(), missions.end(), [](const Mission &m) { return !m.IsValid(); });
+	if(mit != missions.end())
+	{
+		if(ui)
+		{
+			// Prompt the player about any paused missions, but if there are many
+			// do not name them all (since this would overflow the screen).
+			string message = "These active missions or jobs were deactivated due to a missing definition - perhaps you recently removed a plugin?\n";
+			auto it = mit;
+			int named = 0;
+			while(it != missions.end() && (++named < 10))
+			{
+				message += "\t\"" + (*it).Name() + "\"\n";
+				++it;
+			}
+			if(named >= 10 && it != missions.end())
```

Comment:
Just checking for `it != missions.end()` would seem sufficient here, no need to check `named`.